### PR TITLE
Add execution of the examples to test suite

### PR DESCRIPTION
**Pull request by cary-ilm**
This ensure the example code gets included in the coverage analysis

### CODE REVIEW

**Review by meshula — APPROVED** [review]
lgtm, irrespective of windows test fails

**Comment by meshula** [comment]
Are these a new regression?

```
3: =======
[673](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:674)
23: Running testReadScans
[674](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:675)
23: err_cb ERROR 4: Part index (-1) out of range
[675](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:676)
23: err_cb ERROR 4: Part index (11) out of range
[676](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:677)
23: err_cb ERROR 3: Invalid argument to function
[677](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:678)
23: err_cb ERROR 4: Part index (-1) out of range
[678](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:679)
23: err_cb ERROR 4: Part index (11) out of range
[679](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:680)
23: err_cb ERROR 4: Part index (-1) out of range
[680](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:681)
23: err_cb ERROR 4: Part index (1) out of range
[681](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:682)
23: err_cb ERROR 3: NULL output for 'dataWindow'
[682](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:683)
23: err_cb ERROR 4: Part index (-1) out of range
[683](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:684)
23: err_cb ERROR 4: Part index (1) out of range
[684](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:685)
23: err_cb ERROR 3: Invalid argument to function
[685](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:686)
23: err_cb ERROR 19: Attempt to use a tiled accessor function for a scanline image
[686](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:687)
23: err_cb ERROR 3: Invalid request for scanline 19 outside range of data window (20 - 174)
[687](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:688)
23: err_cb ERROR 3: Invalid request for scanline 175 outside range of data window (20 - 174)
[688](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:689)
23: removing temp dir C:\Users\RUNNER~1\AppData\Local\Temp\OpenEXRTest_VCDGOXGG\
[689](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:690)
 23/113 Test  #23: OpenEXRCore.testReadScans .................   Passed    0.01 sec
[690](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:691)
test 24
[691](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:692)
        Start  24: OpenEXRCore.testReadTiles
[692](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:693)

[693](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:694)
24: Test command: D:\a\openexr\openexr\_build\bin\Release\OpenEXRCoreTest.exe "testReadTiles"
[694](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:695)
24: Working Directory: D:/a/openexr/openexr/_build/src/test/OpenEXRCoreTest
[695](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:696)
24: Test timeout computed to be: 7200
[696](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:697)
24: tempDir = 'C:\Users\RUNNER~1\AppData\Local\Temp\OpenEXRTest_VCDGOXGG\': 58
[697](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:698)
24: 
[698](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:699)
24: =======
[699](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:700)
24: Running testReadTiles
[700](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:701)
24: err_cb ERROR 4: Part index (-1) out of range
[701](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:702)
24: err_cb ERROR 4: Part index (11) out of range
[702](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:703)
24: err_cb ERROR 3: Invalid argument to function
[703](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:704)
24: err_cb ERROR 18: Attempt to use a scanline accessor function for a tiled image
[704](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:705)
24: err_cb ERROR 4: Part index (1) out of range
[705](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:706)
24: err_cb ERROR 4: Argument to function out of valid range
[706](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:707)
24: err_cb ERROR 4: Argument to function out of valid range
[707](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:708)
24: err_cb ERROR 4: Argument to function out of valid range
[708](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:709)
24: err_cb ERROR 4: Argument to function out of valid range
[709](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:710)
24: err_cb ERROR 18: Attempt to use a scanline accessor function for a tiled image
[710](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:711)
24: err_cb ERROR 4: Part index (-1) out of range
[711](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:712)
24: err_cb ERROR 4: Part index (1) out of range
[712](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:713)
24: err_cb ERROR 3: Invalid argument to function
[713](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:714)
24: err_cb ERROR 3: Invalid tile indices provided (4, 2, level 0, -1)
[714](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:715)
24: err_cb ERROR 3: Invalid tile indices provided (4, 2, level -1, 0)
[715](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:716)
24: err_cb ERROR 3: Invalid tile indices provided (4, -2, level 0, 0)
[716](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:717)
24: err_cb ERROR 3: Invalid tile indices provided (-4, 2, level 0, 0)
[717](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:718)
24: removing temp dir C:\Users\RUNNER~1\AppData\Local\Temp\OpenEXRTest_VCDGOXGG\
[718](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:719)
 24/113 Test  #24: OpenEXRCore.testReadTiles .................   Passed    0.01 sec
[719](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:720)
test 25
[720](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:721)
        Start  25: OpenEXRCore.testReadMultiPart
[721](https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844#step:6:722)
```

**Comment by peterhillman** [comment]
@meshula I think those particular tests work as intended, and do pass. They check that the core library correctly reports errors when it is passed dumb stuff, like out of range part numbers. @kdt3rd is my theory correct? Maybe it would be better to remove the word 'ERROR' from the output printed by the test suite's err_cb function, so it doesn't confuse by showing up when failures are expected. If a test goes wrong it should print "Return Error" anyway

**Comment by meshula** [comment]
@peterhillman Yeah, maybe rewording would help. I was trying to spot where this test was failing https://github.com/AcademySoftwareFoundation/openexr/actions/runs/4256875505/jobs/7406300844 and I'm afraid maybe my brain was melting by the time I got there, and thought maybe I'd finally found the _actual_ error....

**Comment by darbyjohnston** [comment]
I think the failing test is unable to find the DLLs at runtime, adding this to the CMakeLists.txt should fix it:
```
set_target_properties(OpenEXRExamples PROPERTIES
  RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/bin"
)
```

**Comment by cary-ilm** [comment]
Thanks for this, Darby! I’ll add that fix in and report back.

**Comment by darbyjohnston** [comment]
> [Move RUNTIME_OUTPUT_DIRECTORY inside WIN32](https://github.com/AcademySoftwareFoundation/openexr/pull/1345/commits/1c98b37170081ec93ddaaaec3a7acbea6c78f92f)

I wonder what the difference is, the other tests look like they set the RUNTIME_OUTPUT_DIRECTORY for all systems.

**Comment by cary-ilm** [comment]
The WIN32 thing was a red herring. The CI builds the example code in two different ways, first as a part of the library build, and secondly as an external program linked against the libraries it just built, to mimic an application program. The workflow script executed the program via "./OpenEXRExamples, but setting RUNTIME_OUTPUT_DIRECTORY to bin meant the program is now located in the bin directory.
